pkg/controller/storagecluster/reconcile.go

```diff
@@ -371,17 +371,17 @@ func (r *ReconcileStorageCluster) reconcileNodeTopologyMap(sc *ocsv1.StorageClus
 						updated = true
 					}
 				}
-				if strings.Contains(label, "rack") {
-					if !nodeRacks.Contains(value, node.Name) {
-						nodeRacks.Add(value, node.Name)
-					}
+			}
+			if strings.Contains(label, "rack") {
+				if !nodeRacks.Contains(value, node.Name) {
+					nodeRacks.Add(value, node.Name)
```

Comment:
This could/should still have been a separate commit, as it is a code cleanup/improvement not needed for actual fix that is mentioned in the commit message.